src/Std/Sat/AIG/CachedGatesLemmas.lean

```diff
@@ -0,0 +1,268 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving
+-/
+import Std.Sat.AIG.CachedGates
+import Std.Sat.AIG.LawfulOperator
+
+/-!
+This module contains the theory of the cached gate creation functions, mostly enabled by the
+`LawfulOperator` framework. It is mainly concerned with proving lemmas about the denotational
+semantics of the gate functions in different scenarios.
+-/
+
+namespace Std
+namespace Sat
+
+namespace AIG
+
+/--
+Encoding not as boolean expression in AIG form.
+-/
```

Comment:
`Encoding of not as boolean expression` or `Encode not as boolean expression`

The same comment applies below.